build_projects/dotnet-cli-build/CompileTargets.cs

```diff
@@ -423,16 +226,31 @@ private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli d
             }
 
             // cleanup compilers project output we don't need
-            DeleteMainPublishOutput(outputDir, "compilers");
-            File.Delete(compilersDeps);
-            File.Delete(compilersRuntimeConfig);
-
-            CrossgenUtil.CrossgenDirectory(c, outputDir);
+            PublishMutationUtilties.CleanPublishOutput(
+                sdkOutputDirectory, 
+                "compilers", 
+                deleteRuntimeConfigJson: true, 
+                deleteDepsJson: true);
+
+            // Crossgen SDK directory
+            var sharedFrameworkNugetVersion = DependencyVersions.SharedFrameworkVersion;
+            var sharedFrameworkNameVersionPath = SharedFrameworkPublisher.GetSharedFrameworkPublishPath(
+                rootOutputDirectory,
+                sharedFrameworkNugetVersion);
+            
+            // Copy Host to SDK Directory
+            File.Copy(Path.Combine(sharedFrameworkNameVersionPath, DotnetHostBaseName), Path.Combine(sdkOutputDirectory, $"corehost{Constants.ExeSuffix}"), overwrite: true);
+            File.Copy(Path.Combine(sharedFrameworkNameVersionPath, $"{Constants.DynamicLibPrefix}hostfxr{Constants.DynamicLibSuffix}"), Path.Combine(sdkOutputDirectory, $"{Constants.DynamicLibPrefix}hostfxr{Constants.DynamicLibSuffix}"), overwrite: true);
```

Comment:
I copied these from above but will clean up.

---
In reply to: [64636417](https://github.com/dotnet/cli/pull/3230#discussion_r64636417) [](ancestors = 64636417)